File: socket_client.py
import socket
import sys 

def client_program(port = 5000):
    host = socket.gethostname()  # as both code is running on same pc

    #client_socket = socket.socket()  # instantiate
    try: 
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM) 
    except socket.error as e: 
        print ("Error creating socket: %s" % e) 
        sys.exit(1) 

    try: 
        client_socket.connect((host, port)) 
    except socket.gaierror as e: 
        print ("Address-related error connecting to server: %s" % e) 
        sys.exit(1) 
    except socket.error as e: 
        print ("Connection error: %s" % e) 
        sys.exit(1) 

    print("\nType 'bye' to end \n")

    message = input(" what transactionTime? ")  # take input

    while message.lower().strip() != 'bye':
        try: 
            client_socket.send(message.encode())  # send message
        except socket.error as e: 
            print ("Error sending data: %s" % e) 
            sys.exit(1) 

        try: 
            data = client_socket.recv(1024).decode()  # receive response
        except socket.error as e: 
            print ("Error receiving data: %s" % e) 
            sys.exit(1) 
        

        print('Received from server: ' + data + "\n")  # show in terminal

        message = input(" time again: ?\n ")  # again take input

    client_socket.close()  # close the connection


if __name__ == '__main__':
    port = int (input("enter port number: "))
    client_program(port)

File: socket_client.cpp
// g++ socket_client.cpp -std=c++17 -o client && ./client
#include <iostream>
#include <string>
#include <cstring>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>

#define SERVER_IP "127.0.0.1"
#define PORT 5000
#define BUFFER_SIZE 1024

void client_program(int port)
{
    const char *host = SERVER_IP;

    int client_socket = socket(AF_INET, SOCK_STREAM, 0);
    if (client_socket == -1)
    {
        perror("Error creating socket");
        exit(1);
    }

    sockaddr_in server_addr{};
    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(port);

    if (inet_pton(AF_INET, host, &(server_addr.sin_addr)) <= 0)
    {
        perror("Address-related error connecting to server");
        exit(1);
    }

    if (connect(client_socket, reinterpret_cast<struct sockaddr *>(&server_addr), sizeof(server_addr)) < 0)
    {
        perror("Connection error");
        exit(1);
    }

    std::cout << "\nType 'bye' to end\n\n";

    char buffer[BUFFER_SIZE] = {0};
    std::string message;

    std::cout << "What transactionTime? ";
    std::getline(std::cin, message);

    while (message != "bye")
    {
        if (send(client_socket, message.c_str(), message.size(), 0) == -1)
        {
            perror("Error sending data");
            exit(1);
        }

        memset(buffer, 0, sizeof(buffer));
        if (recv(client_socket, buffer, BUFFER_SIZE, 0) == -1)
        {
            perror("Error receiving data");
            exit(1);
        }
        std::string data(buffer);

        std::cout << "Received from server: " << data << "\n\n";

        std::cout << "TransactionTime again? ";
        std::getline(std::cin, message);
    }

    close(client_socket);
}

int main()
{
    client_program(PORT);
    return 0;
}
